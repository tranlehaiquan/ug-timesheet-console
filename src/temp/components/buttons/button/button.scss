.sked-button {
  @apply sk-leading-normal sk-text-sm sk-font-medium sk-border sk-border-solid sk-px-4 sk-py-2 sk-relative sk-rounded-medium;
  line-height: 1.3;

  &:focus {
    @apply sk-outline-none;
  }

  &--primary {
    @apply sk-bg-blue sk-text-white sk-border-blue sk-tracking-wide;

    &:hover {
      @apply sk-border-blue-dark sk-bg-blue-dark;
    }

    &[disabled] {
      @apply sk-bg-grey-lightest sk-border-grey-lightest;
    }
  }

  &--secondary {
    @apply sk-text-navy-light sk-bg-transparent sk-border-grey-light;
    
    &:hover {
      @apply sk-bg-grey-lightest;
    }

    &[disabled] {
      @apply sk-bg-grey-lightest sk-border-grey-lightest;
    }
  }

  &--transparent {
    @apply sk-text-navy-light sk-bg-transparent sk-border-transparent;

    &:hover {
      @apply sk-text-navy;
    }
  }

  &[disabled] {
    @apply sk-pointer-events-none sk-text-navy-lightest sk-tracking-normal;
  }

  &--compact {
    @apply sk-px-3 sk-py-1;
    line-height: 1.6;
  }
}