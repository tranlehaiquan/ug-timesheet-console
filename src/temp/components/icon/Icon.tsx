import * as React from "react";
import icons from "./iconPaths";
import classNames from "classnames";

export type IconNames = keyof typeof icons;

interface IProps extends React.SVGAttributes<SVGElement> {
  /**
   * Name of the icon
   */
  name: IconNames;
  /**
   * className applied to the containing div
   */
  className?: string;
  /**
   * Size of the icon to override the default (pixels)
   */
  size?: number;
  /**
   * Function to call on click
   */
  onClick?: (event?: React.MouseEvent<SVGElement>) => void;
}

const DEFAULT_SIZE = 18;

/**
 * The Icon component should be used whenever an icon needs to be displayed. It will naturally inherit any text colour but this can be changed by supplied a class directly. Default size of 18px/18px.
 */
export const Icon = ({
  name,
  className,
  size,
  onClick,
  ...otherProps
}: IProps) => {
  const Component = icons[name];
  return (
    <Component
      {...otherProps}
      height={size || DEFAULT_SIZE}
      width={size || DEFAULT_SIZE}
      onClick={onClick}
      className={classNames("sk-flex-shrink-0 sk-fill-current", className)}
    />
  );
};
