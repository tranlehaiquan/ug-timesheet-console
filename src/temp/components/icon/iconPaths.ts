import actions from "./icons/actions.svg";
import activity from "./icons/activity.svg";
import arrowLeft from "./icons/arrow-left.svg";
import arrowUp from "./icons/arrow-up.svg";
import arrowDown from "./icons/arrow-down.svg";
import attachments from "./icons/attachments.svg";
import attribute from "./icons/attribute.svg";
import attributeFilled from "./icons/attribute-filled.svg";
import availability from "./icons/availability.svg";
import bell from "./icons/bell.svg";
import bellAlert from "./icons/bell-alert.svg";
import bookingGrid from "./icons/booking-grid.svg";
import bookmark from "./icons/bookmark.svg";
import calendar from "./icons/calendar.svg";
import calendarNavRight from "./icons/calendar-nav-right.svg";
import calendarNavLeft from "./icons/calendar-nav-left.svg";
import cancel from "./icons/cancel.svg";
import cardSize from "./icons/card-size.svg";
import chat from "./icons/chat.svg";
import chevronUp from "./icons/chevron-up.svg";
import chevronDown from "./icons/chevron-down.svg";
import chevronLeft from "./icons/chevron-left.svg";
import chevronRight from "./icons/chevron-right.svg";
import contact from "./icons/contact.svg";
import copy from "./icons/copy.svg";
import critical from "./icons/critical.svg";
import dashboard from "./icons/dashboard.svg";
import dateRange from "./icons/date-range.svg";
import details from "./icons/details.svg";
import dispatch from "./icons/dispatch.svg";
import edit from "./icons/edit.svg";
import ellipsisVertical from "./icons/ellipsis-vertical.svg";
import exclamation from "./icons/exclamation.svg";
import filter from "./icons/filter.svg";
import filterNew from "./icons/filter-new.svg";
import glasses from "./icons/glasses.svg";
import help from "./icons/help.svg";
import hide from "./icons/hide.svg";
import info from "./icons/info.svg";
import infoFill from "./icons/info-fill.svg";
import jobFields from "./icons/job-fields.svg";
import jobs from "./icons/jobs.svg";
import location from "./icons/location.svg";
import locked from "./icons/locked.svg";
import logout from "./icons/logout.svg";
import map from "./icons/map.svg";
import mapResource from "./icons/map-resource.svg";
import megaphone from "./icons/megaphone.svg";
import minus from "./icons/minus.svg";
import normal from "./icons/normal.svg";
import notes from "./icons/notes.svg";
import pin from "./icons/pin.svg";
import plus from "./icons/plus.svg";
import remove from "./icons/remove.svg";
import restore from "./icons/restore.svg";
import salesforce from "./icons/salesforce.svg";
import search from "./icons/search.svg";
import settings from "./icons/settings.svg";
import skedulo from "./icons/skedulo.svg";
import skeduloText from "./icons/skedulo-text.svg";
import swimlane from "./icons/swimlane.svg";
import swimlaneVertical from "./icons/swimlane-vertical.svg";
import tickCircle from "./icons/tick-circle.svg";
import tick from "./icons/tick.svg";
import time from "./icons/time.svg";
import trash from "./icons/trash.svg";
import view from "./icons/view.svg";
import warning from "./icons/warning.svg";
import notify from "./icons/notify.svg";
import openLink from "./icons/open-link.svg";
import optimise from "./icons/optimise.svg";
import optimiseFill from "./icons/optimise-fill.svg";
import phone from "./icons/phone.svg";
import planeLanding from "./icons/plane-landing.svg";
import planeLandingGround from "./icons/plane-landing-with-ground.svg";
import planeTakeoff from "./icons/plane-take-off.svg";
import push from "./icons/push.svg";
import recurring from "./icons/recurring.svg";
import reorder from "./icons/reorder.svg";
import resource from "./icons/resource.svg";
import resourceAbove from "./icons/resource-above.svg";
import resourceBelow from "./icons/resource-below.svg";
import resourceAdd from "./icons/resource-add.svg";
import resourceRemove from "./icons/resource-remove.svg";
import scheduling from "./icons/scheduling.svg";
import shift from "./icons/shift.svg";
import shiftOvernight from "./icons/shift-overnight.svg";
import sms from "./icons/sms.svg";
import sortAscending from "./icons/sort-ascending.svg";
import sortDescending from "./icons/sort-descending.svg";
import suggest from "./icons/suggest.svg";
import tasks from "./icons/tasks.svg";
import today from "./icons/today.svg";
import unlocked from "./icons/unlocked.svg";
import unschedule from "./icons/unschedule.svg";
import upDown from "./icons/up-down.svg";
import upload from "./icons/upload.svg";
import urgent from "./icons/urgent.svg";
import warningFill from "./icons/warning-fill.svg";
import zoomIn from "./icons/zoom-in.svg";
import zoomOut from "./icons/zoom-out.svg";

export default {
  actions,
  activity,
  arrowLeft,
  arrowUp,
  arrowDown,
  attachments,
  attribute,
  attributeFilled,
  availability,
  bell,
  bellAlert,
  bookingGrid,
  bookmark,
  calendar,
  calendarNavRight,
  calendarNavLeft,
  cancel,
  cardSize,
  chat,
  chevronUp,
  chevronDown,
  chevronLeft,
  chevronRight,
  contact,
  copy,
  critical,
  dashboard,
  dateRange,
  details,
  dispatch,
  edit,
  ellipsisVertical,
  exclamation,
  filter,
  filterNew,
  glasses,
  help,
  hide,
  info,
  infoFill,
  jobFields,
  jobs,
  location,
  locked,
  logout,
  map,
  mapResource,
  megaphone,
  minus,
  normal,
  notes,
  notify,
  openLink,
  optimise,
  optimiseFill,
  phone,
  pin,
  planeLanding,
  planeLandingGround,
  planeTakeoff,
  plus,
  push,
  recurring,
  remove,
  reorder,
  resource,
  resourceAbove,
  resourceAdd,
  resourceBelow,
  resourceRemove,
  restore,
  salesforce,
  scheduling,
  search,
  settings,
  shift,
  shiftOvernight,
  skedulo,
  skeduloText,
  sms,
  sortAscending,
  sortDescending,
  suggest,
  swimlane,
  swimlaneVertical,
  tasks,
  tick,
  tickCircle,
  time,
  today,
  trash,
  unlocked,
  unschedule,
  upDown,
  upload,
  urgent,
  view,
  warning,
  warningFill,
  zoomIn,
  zoomOut,
};
