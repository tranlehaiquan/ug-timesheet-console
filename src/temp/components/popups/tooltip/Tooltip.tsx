import * as React from "react";
import { Position } from "../info-window/info-window-utils";

import {
  InfoWindow,
  IInfoWindowProps,
} from "../info-window/InfoWindow";

import "./tooltip.scss";

export type ITooltipPosition = Position;
type IColorScheme = "dark" | "light";

export interface ITooltipProps extends IInfoWindowProps {
  /**
   * Colour scheme for the tooltip content.  Available options are 'dark' and 'light'
   */
  colorScheme?: IColorScheme;
}

/**
 * Tooltips are an extened version of the info window that triggers on hover by default, delayed slightly and is styled with tooltip styles
 */
export class Tooltip extends React.PureComponent<React.PropsWithChildren<ITooltipProps>, {}> {
  static defaultProps = {
    event: "hover",
    delayShow: 250,
    colorScheme: "dark",
  };

  render() {
    const { colorScheme, children, containerClassName } = this.props;

    const containerClasses = `sked-tooltip sked-tooltip--${colorScheme} ${
      containerClassName || ""
    }`;
    return (
      <InfoWindow {...this.props} containerClassName={containerClasses}>
        {children}
      </InfoWindow>
    );
  }
}
