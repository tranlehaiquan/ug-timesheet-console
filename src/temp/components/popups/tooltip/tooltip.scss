@import '../../../../variables.scss';
@import '../popup.scss';

$tooltip-max-width: 360px;
$tooltip-default-height: 32px;
$tooltip-dark-background-color: $midnight-blue-1 !default;
$tooltip-light-background-color: $white-1 !default;
$tooltip-color: $white-1 !default;
$tooltip-color-inverse: $black-1;
$tooltip-padding: 9px 12px 7px !default;
$tooltip-line-height: 1.2 !default;
$tooltip-letter-spacing: 0.2px !default;

/* purgecss start ignore */
.sked-tooltip {
  .sked-infowindow__content {
    box-sizing: border-box;
    padding: $tooltip-padding;
    border: none;
    box-shadow: $tooltip-box-shadow;

    max-width: $tooltip-max-width !important;
    min-width: 100px;
    min-height: $tooltip-default-height;
    text-align: center;
    line-height: $tooltip-line-height;
    letter-spacing: $tooltip-letter-spacing;
  }

  &--dark {
    .sked-infowindow__content {
      color: $tooltip-color;
      background: $tooltip-dark-background-color;
    }

    .sked-infowindow__triangle {
      background: $tooltip-dark-background-color;
    }
  }

  &--light {
    .sked-infowindow__content {
      color: $tooltip-color-inverse;
      background: $tooltip-light-background-color;
    }

    .sked-infowindow__triangle {
      background: $tooltip-light-background-color;
    }
  }
}
/* purgecss end ignore */
