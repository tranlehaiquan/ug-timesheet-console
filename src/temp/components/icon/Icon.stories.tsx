import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { select, number } from '@storybook/addon-knobs'
import { action } from '@storybook/addon-actions'

import { Icon, IconNames } from './Icon'
import icons from './iconPaths'

const iconNames = Object.keys(icons) as IconNames[]

storiesOf('Icons', module)
  .add('Overview', () => (
    <div className="sk-p-4">
      <h3 className="sk-my-4 sk-ml-2 sk-text-blue">{ iconNames.length } icons</h3>
      <div className="sk-flex sk-flex-wrap sk-items-center">
        { iconNames.map((icon, index) => {
          return (
            <div className="sk-w-32 sk-h-32 sk-m-1 sk-p-4 sk-border-3 sk-border-grey-lighter sk-flex sk-flex-col sk-justify-center sk-items-center sk-text-grey" key={ `${name}-${index}` }>
              <Icon name={ icon } size={ 40 } />
              <span className="sk-pt-4 sk-text-sm sk-text-grey">{ icon }</span>
            </div>
          )
        }) }
      </div>
    </div>
  ))
  .add(
    'Icon',
    () => <Icon name={ select('Icon name', iconNames, iconNames[0]) } size={ number('Size', 18) } onClick={ action() } />
  )
