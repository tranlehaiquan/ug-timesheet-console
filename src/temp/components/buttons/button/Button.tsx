import * as React from 'react'
import classNames from 'classnames'

import { LoadingSpinner } from '../../loader/spinner/LoadingSpinner'
import { IconNames, Icon } from '../../icon/Icon'

export type ButtonTypes = 'primary' | 'secondary' | 'transparent'

export type IButtonProps = {
  /**
   * The type of button to be displayed. Primary | Secondary | Transparent
   */
  buttonType: ButtonTypes
  icon?: IconNames
  /**
   * Pass additional class names
   */
  className?: string
  /**
   * Disable the button
   */
  disabled?: boolean
  /**
   * Compact sized buttonâ€”used when space is tight - allowing more room for content.
   */
  compact?: boolean
  /**
   * Loading button state
   */
  loading?: boolean
  /**
   * Button is active
   */
  active?: boolean
} & React.ButtonHTMLAttributes<HTMLButtonElement>

/**
 * Button component
 */
export const Button = React.forwardRef<HTMLButtonElement, IButtonProps>(
  ({ buttonType, compact, loading, active, icon, children, className, ...otherProps }, ref) => {
    const buttonClassMap: { [key in ButtonTypes]: string } = {
      primary: 'sked-button--primary',
      secondary: 'sked-button--secondary',
      transparent: 'sked-button--transparent'
    }

    const classes = classNames(
      'sked-button',
      buttonClassMap[buttonType],
      {
        'sked-button--compact': compact,
        'sk-pointer-events-none': loading
      },
      {
        'sked-button--active': active
      },
      className
    )

    return (
      <button ref={ref} type="button" {...otherProps} className={classes}>
        <div
          className={classNames({
            'sk-invisible': loading,
            'sk-visible': !loading
          })}
        >
          <div className={classNames('sk-flex sk-items-center')}>
            {icon && <Icon name={icon} className="sked-button__icon" />}
            {children}
          </div>
        </div>
        <div className="sk-inset-0 sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center">
          <LoadingSpinner className={loading ? '' : 'sk-hidden'} size={18} />
        </div>
      </button>
    )
  }
)
