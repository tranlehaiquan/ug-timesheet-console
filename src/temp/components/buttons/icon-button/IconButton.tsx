import * as React from "react";
import classNames from "classnames";

import { Tooltip, ITooltipProps } from "../../popups/tooltip/Tooltip";
import { IconNames } from "../../icon/Icon";
import { Button, IButtonProps } from "../button/Button";

interface IconButtonProps {
  icon: IconNames;
  tooltipContent: ITooltipProps["content"];
  tooltipDelay?: ITooltipProps["delayShow"];
  disableTooltip?: boolean;
}
export type IIconButton = IconButtonProps & Partial<IButtonProps>;

export const IconButton: React.FC<IIconButton> = ({
  icon,
  tooltipContent,
  buttonType,
  className,
  tooltipDelay,
  disableTooltip,
  ...otherProps
}) => {
  const button = (
    <Button
      buttonType={buttonType || "transparent"}
      className={classNames("sked-button--icon-only", className)}
      icon={icon}
      {...otherProps}
    />
  );

  if (disableTooltip) {
    return button;
  }

  return (
    <Tooltip content={tooltipContent} position="top" delayShow={tooltipDelay}>
      {button}
    </Tooltip>
  );
};
