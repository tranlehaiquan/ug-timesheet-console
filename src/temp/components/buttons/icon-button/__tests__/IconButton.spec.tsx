import * as React from 'react'
import { render, shallow, mount } from 'enzyme'

import { Tooltip } from '../../../popups/tooltip/Tooltip'
import { Icon } from '../../../icon/Icon'
import { Button } from '../../button/Button'
import { IconButton } from '../IconButton'

describe('IconButton', () => {
  test('default render', () => {
    // Arrange
    const component = render(<IconButton icon="ellipsisVertical" tooltipContent="Icon tooltip content" />)

    // Assert
    expect(component).toMatchSnapshot()
  })

  test('has tooltip', () => {
    // Arrange
    const component = shallow(<IconButton icon="ellipsisVertical" tooltipContent="Icon tooltip content" />)

    // Assert
    expect(component.find(Tooltip).length).toBe(1)
  })

  test('adds correct class', () => {
    // Arrange
    const component = shallow(<IconButton icon="ellipsisVertical" tooltipContent="Icon tooltip content" />)

    // Assert
    expect(component.find(Button).hasClass('sked-button--icon-only')).toBe(true)
  })

  test('defaults to transparent button', () => {
    // Arrange
    const component = shallow(<IconButton icon="ellipsisVertical" tooltipContent="Icon tooltip content" />)

    // Assert
    expect(component.find(Button).prop('buttonType')).toBe('transparent')
  })

  test('uses correct icon', () => {
    // Arrange
    const component = mount(<IconButton icon="ellipsisVertical" tooltipContent="Icon tooltip content" />)

    // Assert
    expect(component.find(Icon).length).toBe(1)
    expect(component.find(Icon).prop('name')).toEqual('ellipsisVertical')
  })
})
