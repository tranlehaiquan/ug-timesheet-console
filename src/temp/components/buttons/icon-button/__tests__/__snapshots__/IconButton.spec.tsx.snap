// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton default render 1`] = `
<div
  class="sked-infowindow__trigger "
  data-sk-name="sked-infowindow-trigger"
>
  <button
    class="sked-button sked-button--transparent sked-button--icon-only"
    type="button"
  >
    <div
      class="sk-visible"
    >
      <div
        class="sk-flex sk-items-center"
      >
        <svg
          class="sk-flex-shrink-0 sk-fill-current sked-button__icon"
          height="18"
          id="Layer_1"
          viewBox="0 0 18 18"
          width="18"
        >
          <circle
            cx="9"
            cy="15"
            r="2"
          />
          <circle
            cx="9"
            cy="9"
            r="2"
          />
          <circle
            cx="9"
            cy="3"
            r="2"
          />
        </svg>
      </div>
    </div>
    <div
      class="sk-inset-0 sk-absolute sk-m-auto sk-flex sk-justify-center sk-items-center"
    >
      <div
        class="loading-spinner sk-hidden"
      >
        <svg
          class="loading-spinner--circular"
          height="18"
          viewBox="25 25 50 50"
          width="18"
        >
          <circle
            class="loading-spinner--path"
            cx="50"
            cy="50"
            fill="none"
            r="20"
            stroke="currentColor"
            stroke-miterlimit="10"
            stroke-width="5"
          />
        </svg>
      </div>
    </div>
  </button>
</div>
`;
